test/e2e/azure_machinepool_drain.go

```diff
@@ -157,13 +166,39 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 			return err
 		}
 
-		decreasedReplicas := *owningMachinePool.Spec.Replicas - int32(1)
+		decreasedReplicas = *owningMachinePool.Spec.Replicas - int32(1)
 		owningMachinePool.Spec.Replicas = &decreasedReplicas
 		return helper.Patch(ctx, owningMachinePool)
 	}, 3*time.Minute, 3*time.Second).Should(Succeed())
 
-	// TODO setup a watcher to validate expected 2nd order drain outcomes
-	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159
+	By(fmt.Sprintf("checking for a machine to start draining for machine pool: %s/%s", amp.Namespace, amp.Name))
+	Eventually(func() error {
+		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
+		if err != nil {
+			LogWarning(errors.Wrap(err, "failed to list the azure machine pool machines").Error())
+			return err
+		}
+
+		for i := range ampmls {
```

Comment:
can we use the human-friendly var instead of the index int?

`for _, ampm := range ampls {`...

Also, I think we should be able to simply check for the `IsTrue` condition below, as it is inclusive of `conditions.Has`. So just:

`if conditions.IsTrue(&ampm, clusterv1.DrainingSucceededCondition) {`...

should work

Reply:
> can we use the human-friendly var instead of the index int?
> 
> `for _, ampm := range ampls {`...

Unfortunately, the linter complains about referencing a temporarily for loop variable if I do this.

Reply:
I've learned to just do what the linter says